cypress/component/column-linking-table.cy.js

```diff
@@ -0,0 +1,228 @@
+
+// Documentation for testing Vue events of components
+// https://docs.cypress.io/guides/component-testing/events-vue
+
+// Mocks
+
+const store = {
+
+    state: {
+
+        categories: [
+
+            "Subject ID",
+            "Age",
+            "Sex",
+            "Diagnosis",
+            "Assessment Tool"
+        ],
+
+        categoryClasses: {
+
+            "Subject ID": "category-style-0",
+            "Age": "category-style-1",
+            "Sex": "category-style-2",
+            "Diagnosis": "category-style-3",
+            "Assessment Tool": "category-style-4"
+        },
+
+        columnToCategoryMap: {
+
+            "participant_id": null,
+            "age": null,
+            "sex": null,
+            "group": null,
+            "group_dx": null,
+            "number_comorbid_dx": null,
+            "medload": null,
+            "iq": null,
+            "session": null
+        },
+
+        dataDictionary: {
+
+            original: {
+
+                "age": { "Description": "age of the participant" },
+                "sex": { "Description": "sex of the participant as reported by the participant" },
+                "group": { "Description": "diagnostic status determined by the study clinician at baseline" },
+                "group_dx": { "Description": "specific diagnosis determined by the study clinician at baseline" },
+                "number_comorbid_dx": { "Description": "a number of diagnoses comorbid with UD (e.g., GAD, PTSD)" },
+                "medload": { "Description": "reflects the number of dosage of psychotropic medications taken by participants. Higher numbers correspond to more medications and/or higher medication dosage " },
+                "iq": { "Description": "IQ derived based on the NART assessment." },
+                "session": { "Description": "scanning session" }
+            }
+        },
+
+        dataTable: {
+
+            columns: [
+
+                "participant_id",
+                "age",
+                "sex",
+                "group",
+                "group_dx",
+                "number_comorbid_dx",
+                "medload",
+                "iq",
+                "session"
+            ],
+
+            original: [{
+
+                "participant_id": "",
+                "age": "",
+                "sex": "",
+                "group": "",
+                "group_dx": "",
+                "number_comorbid_dx": "",
+                "medload": "",
+                "iq": "",
+                "session": ""
+            }]
+        },
+
+        toolColorPalette: {
+
+            color1: "category-style-0",
+            color2: "category-style-1",
+            color3: "category-style-2",
+            color4: "category-style-3",
+            color5: "category-style-4",
+            colorDefault: "category-style-default"
+        }
+    },
+
+    actions: {
+
+        dispatch: (p_actionName, p_payload) => { }
```

Comment:
I really like this. I haven't added the actions to my tests but I think it helps to document them explicitly. One thing though: shouldn't there be two actions here, `linkColumnWithCategory` and `unlinkColumnFromCategory`?

Reply:
`linkColumnWithCategory` and `unlinkColumnFromCategory` will be converted to mutations and a new action `alterColumnCategoryRelation` has been added that will call (in future store refactor) either depending on the given action input and state of the store field `columnToCategoryMap`.